options in query

beowulfenator opened the pull request:
This is regarding issue #70.

cebe commented on Query.php:
```diff
@@ -157,6 +157,10 @@ class Query extends Component implements QueryInterface
     public $minScore;
+    /**
+     * @var array list of options that will passed to commands created by this query.
+     */
```
need `@since` annotation.

cebe commented on Query.php:
```diff
@@ -607,4 +611,20 @@ public function minScore($minScore)
+     * Sets the options to be passed to the command created by this query.
+     * @param array $options the options to be set.
+     * @return $this the query object itself
+     */
```
same here.

cebe commented on Query.php:
```diff
@@ -607,4 +611,20 @@ public function minScore($minScore)
     }
+
+    /**
+     * Sets the options to be passed to the command created by this query.
```
should be more clear about what kind of options these are, e.g. `options that are appended to the query URL such as preference, ... etc...`

cebe commented:
Travis CI fails, maybe the version does not support the options? https://travis-ci.org/yiisoft/yii2-elasticsearch/jobs/112157332#L47

beowulfenator commented:
I'm glad you agree with this in general. I'll add more docs, tests, and probably the `addOptions()` method. I wonder why it's failing the test, though. Will investigate.

beowulfenator commented:
The tests were failing, because the option that I added as a test was not supported in older versions of ES. I've changed the test to use another option, and all is well now.

@SilverFire, I've fixed tests for minScore. Minimum score is supported in all versions of ES, but to test it we need to explicitly assign scores to test results, which I do using `script_score`.

@cebe, any other suggestions?

cebe commented on Query.php:
```diff
@@ -607,4 +613,41 @@ public function minScore($minScore)
+    public function options($options)
+    {
+        if (!is_array($options)) {
+            throw new \yii\base\InvalidParamException('Array parameter expected, ' . gettype($options) . ' received.');
```
`\yii\base\InvalidParamException` should be included with `use`.

cebe commented:
If you fix the exception class `use` and add a CHANGELOG line it is good to merge imo

cebe commented:
:+1:

SilverFire commented on Query.php:
```diff
@@ -157,6 +158,12 @@ class Query extends Component implements QueryInterface
      */
     public $minScore;
+    /**
+     * @var array list of options that will passed to commands created by this query.
```
`will passed` -> `will be passed`
